hashicorp/terraform-provider-tfe
# Add `pull_request_outputs_enabled` on workspace settings

## Description

Note: This PR is still a draft and is not ready to be merged yet.

This PR adds the `pull_request_outputs_enabled` field on workspace settings.

_Remember to:_

- _Update the [Change Log](https://github.com/hashicorp/terraform-provider-tfe/blob/main/CONTRIBUTING.md#updating-the-changelog)_

- _Update the [Documentation](https://github.com/hashicorp/terraform-provider-tfe/blob/main/CONTRIBUTING.md#updating-the-documentation)_

## Testing plan

1.  Attempt to create a workspace with invalid configuration, where `pull_request_outputs_enabled` is not allowed to be `true` when `speculative_enabled` is set to `false`. Expect an error to be thrown.
```
resource "tfe_organization" "foobar" {
  name  = "tst-terraform-1"
  email = "admin@company.com"
}

resource "tfe_workspace" "foobar" {
  name                         = "workspace-test"
  organization                 = tfe_organization.foobar.id
  speculative_enabled          = false
  pull_request_outputs_enabled = true
}
```
2. Create a workspace with valid configuration, where `pull_request_outputs_enabled` is `false` and  `speculative_enabled` is `true`.
3. Turn on `pull_request_outputs_enabled` for the existing workspace.
4. Attempt to turn off `pull_request_outputs_enabled` while `speculative_enabled` is `true`. Expect an error to be thrown.


## External links
- [go-tfe PR](https://github.com/hashicorp/go-tfe/pull/486)
- [go-tfe documentation](https://pkg.go.dev/github.com/hashicorp/go-tfe#Workspace)
- [API documentation](https://www.terraform.io/cloud-docs/api-docs/workspaces)

## Output from acceptance tests
```
> TESTARGS="-run TestAccTFEWorkspace_pullRequestOutputsEnabled" make testacc
TF_ACC=1 TF_LOG_SDK_PROTO=OFF go test $(go list ./... |grep -v 'vendor') -v -run TestAccTFEWorkspace_pullRequestOutputsEnabled -timeout 15m
?       github.com/hashicorp/terraform-provider-tfe     [no test files]
=== RUN   TestAccTFEWorkspace_pullRequestOutputsEnabled
--- PASS: TestAccTFEWorkspace_pullRequestOutputsEnabled (36.94s)
PASS
ok      github.com/hashicorp/terraform-provider-tfe/tfe 37.288s
?       github.com/hashicorp/terraform-provider-tfe/version     [no test files]
```

```
> TESTARGS="-run TestAccTFEWorkspaceDataSource_basic" make testacc
TF_ACC=1 TF_LOG_SDK_PROTO=OFF go test $(go list ./... |grep -v 'vendor') -v -run TestAccTFEWorkspaceDataSource_basic -timeout 15m
?       github.com/hashicorp/terraform-provider-tfe     [no test files]
=== RUN   TestAccTFEWorkspaceDataSource_basic
--- PASS: TestAccTFEWorkspaceDataSource_basic (11.47s)
PASS
ok      github.com/hashicorp/terraform-provider-tfe/tfe 11.829s
?       github.com/hashicorp/terraform-provider-tfe/version     [no test files]
```

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202331487321513
